test/ci-operator-configresolver-integration/run.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# This test runs the ci-operator-configresolver and verifies that it returns
+# properly resolved configs
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ROOTDIR=$(pwd)
+WORKDIR="$( mktemp -d )"
+trap "rm -rf ${WORKDIR}" EXIT
+
+pushd $WORKDIR
+go build -o tester $ROOTDIR/test/ci-operator-configresolver-integration/main.go
+# copy registry to tmpdir to allow tester to modify registry
+cp -a $ROOTDIR/test/ci-operator-configresolver-integration/ tests
+ci-operator-configresolver -config tests/configs -log-level debug &
+PID=$!
+# wait for registry to be resolved
+sleep 1
```

Comment:
nit: I am not feeling comfortable to rely on a sleep command to wait when the service is ready to accept connections. Maybe it is better to expose another route just for it.

Reply:
The main issue is that, while the loading of the config files and exposing the service is really quick, it takes slightly longer than making the actual GET request. I think maybe having a loop with a timeout might work. We could have it attempt to get the proper config, and then try every second for 5 seconds, failing if the output is still not correct.

We'll probably need to do something like that when we add automatic reloading as well, as we might get a stale result from just immediately checking the output after updating a registry/config file.

Reply:
Using the upstream libraries to expose a health and readiness endpoint and using that to gauge when the service is ready to accept the request is preferred, but can be a follow-up.

Reply:
While using exposing health and readiness endpoints can be good to indicate whether or not the server is ready to serve configs, it doesn't help much for the tests checking if the server picked up changes made on disk. The server will still be ready; it will just serve outdated information until it detects the changes and updates the internal mappings.